lib/travis/build/script/node_js.rb

```diff
@@ -47,15 +48,52 @@ def cache_slug
 
         private
 
+          def convert_legacy_nodejs_config
+            # TODO deprecate :nodejs
+            # some old projects use language: nodejs. MK.
+            if config[:nodejs] && !config[:node_js]
+              config[:node_js] = config[:nodejs]
+            end
+          end
+
+          def node_js?
+            !!config[:node_js]
```

Comment:
I think our insistence to reinstate the `DEFAULTS` hash is the source of the spec failures in https://travis-ci.org/travis-ci/travis-build/builds/70773325.

In the case of RVM and Ruby, we set the version in the `DEFAULTS` hash to be `default`, which RVM accepts; if that is the Ruby version we see in `travis-build` & the file `.ruby-version` exists, we use it to set the Ruby version.

Removing `DEFAULTS` will make the failing specs in https://travis-ci.org/travis-ci/travis-build/builds/70773325 pass.

The problem here is that, without the `DEFAULT` hash and the `node_js` value present in `.travis.yml`, the environment variable `$TRAVIS_NODE_VERSION` will not be set. And the spec checking that will fail. I am not sure yet how we can reconcile all of these issues.

(Incidentally, `$TRAVIS_RUBY_VERSION` will be set to `default` if `.travis.yml` does not define the Ruby version for Ruby builds. This, too, is a problem.)